homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -139,19 +245,44 @@ def device_state_attributes(self):
             _LOGGER.debug("Got new state: %s", state)
 
             self._available = True
-            self._state = state.aqi
-            self._state_attrs.update(
-                {
-                    ATTR_POWER: state.power,
-                    ATTR_CHARGING: state.usb_power,
-                    ATTR_BATTERY_LEVEL: state.battery,
-                    ATTR_DISPLAY_CLOCK: state.display_clock,
-                    ATTR_NIGHT_MODE: state.night_mode,
-                    ATTR_NIGHT_TIME_BEGIN: state.night_time_begin,
-                    ATTR_NIGHT_TIME_END: state.night_time_end,
-                    ATTR_SENSOR_STATE: state.sensor_state,
-                }
-            )
+
+            if self._model == MODEL_XIAOMI_AIRQUALITYMONITOR_S1:
+                self._state = state.pm25
+                self._state_attrs.update(
```

Comment:
Don't use state attributes to hold the measurements. Each entity should just return the correct measurement as entity state.